llama_index/vector_stores/mongodb.py

```diff
@@ -0,0 +1,191 @@
+        insert_kwargs (Optional[Dict]): kwargs used during `insert`.
+
+    """
+
+    stores_text: bool = True
+
+    def __init__(
+        self,
+        mongodb_client: Optional[Any] = None,
+        db_name: str = "default_db",
+        collection_name: str = "default_collection",
+        index_name: str = "default",
+        id_key: str = "id",
+        embedding_key: str = "embedding",
+        text_key: str = "text",
+        metadata_key: str = "metadata",
+        insert_kwargs: Optional[Dict] = None,
+        **kwargs: Any,
+    ) -> None:
+        """Initialize params."""
+        import_err_msg = "`pymongo` package not found, please run `pip install pymongo`"
+        try:
+            import pymongo  # noqa: F401
+        except ImportError:
+            raise ImportError(import_err_msg)
+
+        if mongodb_client is not None:
+            self._mongodb_client = cast(pymongo.MongoClient, mongodb_client)
+        else:
+            if "MONGO_URI" not in os.environ:
+                raise ValueError(
+                    "Must specify MONGO_URI via env variable "
+                    "if not directly passing in client."
+                )
+            self._mongodb_client = pymongo.MongoClient(os.environ["MONGO_URI"])
+
+        self._collection = self._mongodb_client[db_name][collection_name]
+        self._index_name = index_name
+        self._embedding_key = embedding_key
+        self._id_key = id_key
+        self._text_key = text_key
+        self._metadata_key = metadata_key
+        self._insert_kwargs = insert_kwargs or {}
+
+    def add(
+        self,
+        embedding_results: List[NodeWithEmbedding],
+    ) -> List[str]:
+        """Add embedding results to index.
+
+        Args
+            embedding_results: List[NodeWithEmbedding]: list of embedding results
+
+        """
+        ids = []
+        data_to_insert = []
+        for result in embedding_results:
+            node_id = result.id
+            node = result.node
+
+            metadata = node_to_metadata_dict(node)
+
+            entry = {
+                self._id_key: node_id,
+                self._embedding_key: result.embedding,
+                self._text_key: node.text or "",
+                self._metadata_key: metadata,
+            }
+            data_to_insert.append(entry)
+            ids.append(node_id)
+        logger.debug("Inserting data into MongoDB: %s", data_to_insert)
+        insert_result = self._collection.insert_many(
+            data_to_insert, **self._insert_kwargs
+        )
+        logger.debug("Result of insert: %s", insert_result)
+        return ids
+
+    def delete(self, ref_doc_id: str, **delete_kwargs: Any) -> None:
+        """
+        Delete nodes using with ref_doc_id.
+
+        Args:
+            ref_doc_id (str): The doc_id of the document to delete.
+
+        """
+        # delete by filtering on the doc_id metadata
+        self._collection.delete_one(filter={"id": ref_doc_id}, **delete_kwargs)
```

Comment:
does this actually delete by ref_do_id? It seems like the `id` field has the node_id

Reply:
Should be fixed now

Reply:
Yea this makes more sense, thanks!